src/views/Tenant/TenantView.js

```diff
@@ -0,0 +1,264 @@
+  ];
+  const [activeTab, setActiveTab] = useState(tabs[0].id);
+
+  /**
+   * Handle activating edit form
+   */
+  const handleEditToggle = () => setEditingStatus(!isEditing);
+  const onFormikSubmit = (values, { setSubmitting }) => {
+    setSubmitting(true);
+    setState({
+      ...state,
+      firstName: values.firstName,
+      lastName: values.lastName,
+      phone: values.phone,
+      email: values.email,
+    });
+    setTimeout(() => {
+      setSubmitting(false);
+      setEditingStatus(false);
+    }, 500);
+  };
+
+  /**
+   * Handle press cancel button
+   */
+  const onCancelClick = () => {
+    setEditingStatus(false);
+  };
+
+  /**
+   * Convert an array of staff to an array of SearchPanelItems
+   * @param {*} staffArray
+   */
+  const getStaffChoice = (staffArray) => {
+    const staffChoices = [];
+    if (staffArray && Array.isArray(staffArray)) {
+      staffArray.forEach((staff) => {
+        const name = `${staff.firstName} ${staff.lastName}`;
+        const staffChoice = { key: staff.id, description: name };
+        staffChoices.push(staffChoice);
+      });
+    }
+    return staffChoices;
+  };
+
+  /**
+   * Get a tenant
+   */
+  const getTenant = async () => {
+    const tenantResponse = await axios.get(`/api/tenants/${id}`);
+    const tenant = tenantResponse.data;
+    const propertyUrl = `/api/properties/${tenant.propertyName}`;
+    const propertyResponse = await axios.get(propertyUrl);
+    const property = propertyResponse.data;
+    const ticketsResponse = await axios.get(`/api/tickets?tenant=${tenant.id}`);
+    const tickets = ticketsResponse.data;
+    setState({ tenant, property, tickets });
+
+    const currentStaff = getStaffChoice(tenant.staff);
+
+    setStaffSelections(currentStaff);
+  };
+
+  /**
+   * When component mounts, get tenant.
+   */
+  useEffect(() => {
+    getTenant();
+  }, []);
+
+  const { tenant } = state;
+  const { property } = state;
+
+  /**
+   * Configure edit table
+   */
+  const getTableData = () => [
+    {
+      key: "firstName",
+      label: "First Name",
+      value: tenant.firstName,
+      inputType: "text",
+    },
+    {
+      key: "lastName",
+      label: "Last Name",
+      value: tenant.lastName,
+      inputType: "text",
+    },
+    {
+      key: "phone",
+      label: "Phone",
+      value: tenant.phone,
+      inputType: "text",
+    },
+    {
+      key: "address",
+      label: "Property",
+      value: `${property.address}, ${property.city}, ${property.state}, ${property.zipcode}`,
+      inputType: "text",
+      comp: <div />,
+    },
+    {
+      key: "unit",
+      label: "Unit",
+      value: property.unit,
+      inputType: "text",
+      comp: <div />,
+    },
+  ];
+
+  /**
+   * When staff search input text changes, call API to find matching users.
+   */
+  useEffect(() => {
+    const loadStaff = async () => {
+      const staffResponse = await axios.post("/api/users/role", { userrole: 2, name: staffSearchText });
```

Comment:
So Join Staff is userrole 3 while property managers is userrole 2

Reply:
I'll fix that and add a constant for it, so this isn't a magic number.

Reply:
I changed it to 3, but it looks like we don't have any seed data for STAFF. So, this might be tough to test again without that. You can observe the API requests in developer tools, though.